`tests/_models/test_dag.py::test_package_dag_from_pkgs_uses_pep503normalize` fails if `flufl.lock` is installed

If `flufl.lock` package is installed in the test environment, the aforementioned test fails:

```pytb
$ python -m pytest tests/_models/test_dag.py::test_package_dag_from_pkgs_uses_pep503normalize
========================================================= test session starts =========================================================
platform linux -- Python 3.10.13, pytest-8.0.2, pluggy-1.4.0
rootdir: /tmp/pipdeptree
plugins: mock-3.12.0, cov-4.1.0
collected 1 item                                                                                                                      

tests/_models/test_dag.py F                                                                                                     [100%]

============================================================== FAILURES ===============================================================
___________________________________________ test_package_dag_from_pkgs_uses_pep503normalize ___________________________________________

mock_pkgs = <function mock_pkgs.<locals>.func at 0x7f88f2a83640>

    def test_package_dag_from_pkgs_uses_pep503normalize(mock_pkgs: Callable[[MockGraph], Iterator[Mock]]) -> None:
        # ensure that requirement gets matched with a dists even when it's key needs pep503 normalization to match
        graph: dict[tuple[str, str], list[tuple[str, list[tuple[str, str]]]]] = {
            ("parent-package", "1.2.3"): [("flufl.lock", [(">=", "2.0.0")])],
            ("flufl-lock", "2.2.0"): [],
        }
        package_dag = PackageDAG.from_pkgs(list(mock_pkgs(graph)))
        parent_key = "parent-package"
        c = package_dag.get_children(parent_key)
        assert c[0].dist
>       assert c[0].project_name == "flufl-lock"
E       AssertionError: assert 'flufl.lock' == 'flufl-lock'
E         
E         - flufl-lock
E         ?      ^
E         + flufl.lock
E         ?      ^

tests/_models/test_dag.py:119: AssertionError
========================================================== warnings summary ===========================================================
.tox/py310/lib/python3.10/site-packages/pipdeptree/_models/package.py:10
  /tmp/pipdeptree/.tox/py310/lib/python3.10/site-packages/pipdeptree/_models/package.py:10: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    from pip._vendor.pkg_resources import Requirement  # noqa: PLC2701

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================= short test summary info =======================================================
FAILED tests/_models/test_dag.py::test_package_dag_from_pkgs_uses_pep503normalize - AssertionError: assert 'flufl.lock' == 'flufl-lock'
==================================================== 1 failed, 1 warning in 0.11s =====================================================
```

Easy reproducer:

```
tox -e py310
. .tox/py310/bin/activate
pip install flufl-lock
python -m pytest tests/_models/test_dag.py::test_package_dag_from_pkgs_uses_pep503normalize
```

--- a/tests/_models/test_dag.py
+++ b/tests/_models/test_dag.py
@@ -1,15 +1,15 @@
 from __future__ import annotations
 
+from importlib.metadata import PackageNotFoundError
 from itertools import chain
 from typing import TYPE_CHECKING, Any, Callable, Iterator
+from unittest.mock import Mock
 
 import pytest
 
 from pipdeptree._models import DistPackage, PackageDAG, ReqPackage, ReversedPackageDAG
 
 if TYPE_CHECKING:
-    from unittest.mock import Mock
-
     from tests.our_types import MockGraph
 
 
@@ -92,9 +92,16 @@ def test_package_dag_reverse(example_dag: PackageDAG) -> None:
     assert all(isinstance(v, ReqPackage) for v in chain.from_iterable(t2.values()))
 
 
-def test_package_dag_from_pkgs(mock_pkgs: Callable[[MockGraph], Iterator[Mock]]) -> None:
+def test_package_dag_from_pkgs(
+    mock_pkgs: Callable[[MockGraph], Iterator[Mock]], monkeypatch: pytest.MonkeyPatch
+) -> None:
     # when pip's _vendor.packaging.requirements.Requirement's requires() gives a lowercased package name but the actual
     # package name in PyPI is mixed case, expect the mixed case version
+
+    # Since DistPackage.project_name will try to use importlib.metadata to grab the project name, let's avoid using
+    # the environment and use the fallback project name (since we control it here anyways).
+    monkeypatch.setattr("pipdeptree._models.package.metadata", Mock(side_effect=PackageNotFoundError()))
+
     graph: dict[tuple[str, str], list[tuple[str, list[tuple[str, str]]]]] = {
         ("examplePy", "1.2.3"): [("hellopy", [(">=", "2.0.0")])],
         ("HelloPy", "2.2.0"): [],
@@ -106,8 +113,13 @@ def test_package_dag_from_pkgs(mock_pkgs: Callable[[MockGraph], Iterator[Mock]])
     assert c[0].project_name == "HelloPy"
 
 
-def test_package_dag_from_pkgs_uses_pep503normalize(mock_pkgs: Callable[[MockGraph], Iterator[Mock]]) -> None:
+def test_package_dag_from_pkgs_uses_pep503normalize(
+    mock_pkgs: Callable[[MockGraph], Iterator[Mock]], monkeypatch: pytest.MonkeyPatch
+) -> None:
     # ensure that requirement gets matched with a dists even when it's key needs pep503 normalization to match
+
+    monkeypatch.setattr("pipdeptree._models.package.metadata", Mock(side_effect=PackageNotFoundError()))
+
     graph: dict[tuple[str, str], list[tuple[str, list[tuple[str, str]]]]] = {
         ("parent-package", "1.2.3"): [("flufl.lock", [(">=", "2.0.0")])],
         ("flufl-lock", "2.2.0"): [],

